src/Modules/ContentBlocks/Domain/ContentBlock/Command/DeleteContentBlockHandler.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Command;
+
+use ForkCMS\Core\Domain\MessageHandler\CommandHandlerInterface;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
+use Exception;
+
+final class DeleteContentBlockHandler implements CommandHandlerInterface
+{
+    public function __construct(
+        private readonly ContentBlockRepository $contentBlockRepository,
+        private readonly BlockRepository $blockRepository
+    ) {
+    }
+
+    public function __invoke(DeleteContentBlock $deleteContentBlock)
+    {
+        $versions = $this->contentBlockRepository->getVersionsForRevisionId($deleteContentBlock->id);
+
+        if (empty($versions)) {
+            return;
+        }
+
+        $this->contentBlockRepository->removeMultiple($versions);
+
+        try {
+            $extraId = $versions[0]->getExtraId();
+            $block = $this->blockRepository->findOneBy(['id' => $extraId]);
+            if ($block !== null) {
```

Comment:
can you use an instanceof here?

Reply:
https://github.com/forkcms/forkcms/pull/3555/commits/c24fac573aad3d9e429b9a352c24087b2dd1f862